SPARQLWrapper/SmartWrapper.py

```diff
@@ -212,41 +211,57 @@ def __getitem__(self, key):
         :rtype: array of variable -> :class:`Value`  dictionaries
         """
 
-        def _checkKeys(keys):
+        def _checkKeys(keys: Union[List[Any], Tuple[Any, ...]]) -> bool:
             if len(keys) == 0:
                 return False
             for k in keys:
-                if not isinstance(k, str) or k not in self.variables:
+                if (
+                    not isinstance(k, str)
+                    or self.variables is None
+                    or k not in self.variables
+                ):
                     return False
             return True
 
-        def _nonSliceCase(key):
-            if isinstance(key, str) and key != "" and key in self.variables:
-                # unicode or string:
-                return [key]
-            elif type(key) is list or type(key) is tuple:
+        def _nonSliceCase(
+            key: Union[
+                str,
+                slice,
+                List[Any],
+                Tuple[Any],
+            ]
+        ) -> Tuple[bool, Union[List[Any], str]]:
+            if isinstance(key, str):
+                if key != "" and self.variables is not None and key in self.variables:
+                    # unicode or string:
+                    return True, [key]
+            elif isinstance(key, list):
                 if _checkKeys(key):
-                    return key
-            return False
+                    return True, key
+            elif isinstance(key, tuple):
+                if _checkKeys(key):
+                    return True, list(key)
+
+            return False, str(type(key))
```

Comment:
Not entirely sure, but it looks like this can go into a separate PR.

Reply:
934e334